x/interchainstaking/keeper/intent.go

```diff
@@ -136,14 +136,17 @@ func (k *Keeper) AggregateIntents(ctx sdk.Context, zone *types.Zone) error {
 	}
 
 	// normalise aggregated intents again.
-	for idx, intent := range aggregate.Sort() {
-		intent.Weight = intent.Weight.Quo(ordinalizedIntentSum)
-		aggregate[idx] = intent
+	newAggregate := make(types.ValidatorIntents, 0)
```

Comment:
Creating this with `len(aggregate)` and using an idx to set the values is much more performant...
Go slice allocation is rather slow, so if this is being used a lot setting it to the correct size initially will be very beneficial.

Reply:
We don't know the correct size ahead of time, as we are filtering out zero values.

Performance is less critical as this get called once per epoch.